packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -398,7 +402,10 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin {
             vsync: this,
             hasFocus: _focused,
             hovering: _hovering,
-            borderRadius: widget.borderRadius,
+            textDirection: widget.textDirection ?? themeData.checkboxTheme.textDirection ?? TextDirection.ltr,
```

Comment:
The default value of this property should be `Directionality.of(context) ?? TextDirection.ltr`